### PR TITLE
[RFC] [POC] [Cart] Remove SessionCartSubscriber

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CartBundle/Controller/CartItemController.php [hunk 1]
```diff
@@ -137,6 +140,30 @@ public function removeAction(Request $request)
...
 
     /**
+     * @param Request $request
+     * @param CartInterface $cart
+     */
+    protected function saveCartInSession(Request $request, CartInterface $cart)
+    {
+        $session = $request->getSession();
+
+        if (!$session->has($this->getSessionCartPrefix())) {
```
Comment (pamil) [inline src/Sylius/Bundle/CartBundle/Controller/CartItemController.php]: Are you sure? `$session->has('_sylius.cart')` should always return `false` as you do `$session->set('_sylius.cart.CHANNEL_CODE', ...)` instead.

Comment (lchrusciel) [inline src/Sylius/Bundle/CartBundle/Controller/CartItemController.php]: Resolved. :shipit: 

BTW. I hate new github review style...

File: src/Sylius/Bundle/CartBundle/Controller/CartItemController.php [hunk 2]
```diff
@@ -137,6 +140,32 @@ public function removeAction(Request $request)
...
+     * @param Request $request
+     * @param CartInterface $cart
+     */
+    protected function saveCartInSession(Request $request, CartInterface $cart)
+    {
+        $session = $request->getSession();
+
+        if (!$session->has($this->getSessionCartKey($cart))) {
+            $session->set(
+                sprintf('%s.%s', $this->getSessionCartKey($cart), $cart->getChannel()->getCode()),
```
Comment (pamil) [inline src/Sylius/Bundle/CartBundle/Controller/CartItemController.php]: After the changes we're creating here `_sylius.cart.CHANNEL_CODE.CHANNEL_CODE`.

File: src/Sylius/Bundle/CartBundle/Controller/CartItemController.php [hunk 3]
```diff
@@ -137,6 +140,32 @@ public function removeAction(Request $request)
...
+            );
+        }
+    }
+
+    /**
+     * @param CartInterface $cart
+     *
+     * @return string
+     */
+    protected function getSessionCartKey(CartInterface $cart)
```
Comment (pamil) [inline src/Sylius/Bundle/CartBundle/Controller/CartItemController.php]: If cart is per-channel, we'd better pass only the channel to make it clearer.

File: src/Sylius/Bundle/CartBundle/Controller/CartItemController.php (hunk 2 above)
Comment (lchrusciel) [inline src/Sylius/Bundle/CartBundle/Controller/CartItemController.php]: ssshhhiiiieeett.... I have coupled it to the channel bundle. I have to reorganize this...

**Comment by lchrusciel** [comment]
Closing as it has to be redesigned

**Pull request closed without merging**